### PR TITLE
Feature improve handling of known libraries

### PR DESCRIPTION
**Pull request by ciechowoj**
Extends handling of the macro definition dependencies beyond `limits.h`.
Fixes problem with cycles in headers (`throw new Exception("The include graph isn't a DAG.");` has been triggered...).

This PR depends on another PR:
https://github.com/jacob-carlborg/dstep/pull/176

This PR is a step forward to solve:
https://github.com/jacob-carlborg/dstep/issues/141

### CODE REVIEW

**Comment by jacob-carlborg** [comment]
#176 is merged, please rebase.

**Comment by ciechowoj** [comment]
Rebased.

File: dstep/translator/HeaderIndex.d [hunk 1]
```diff
@@ -170,98 +223,170 @@ class HeaderIndex
 {
     private IncludeGraph includeGraph_;
     private string[string] stdLibPaths;
+    private string[string] knownModules;
     private string mainFilePath;
 
     public this(TranslationUnit translationUnit)
     {
+        immutable string[string] standardModuleMapping = [
```
Comment (jacob-carlborg) [inline dstep/translator/HeaderIndex.d]: Shouldn't this be moved out of the constructor as a constant. The same associative array will be created each time the constructor is called, unless the compiler is doing something smart.

Comment (ciechowoj) [reply dstep/translator/HeaderIndex.d]: I tried, but D is too dumb for this : (:

```
dstep/translator/HeaderIndex.d(222,50): Error: non-constant expression `["assert.h":"core.stdc.assert_", 
```

Comment (jacob-carlborg) [reply dstep/translator/HeaderIndex.d]: Hmm, right. One would have to use a shared module constructor for this.

Comment (ciechowoj) [reply dstep/translator/HeaderIndex.d]: Lets keep it simple. It will be called only once per translation unit.

**Comment by jacob-carlborg** [comment]
👌.

**Comment by jacob-carlborg** [comment]
Thanks.

**Pull request merged**